### PR TITLE
CIV-15571 remove notification and task list to inactive when the case go to dec…

### PR DESCRIPTION
**Pull request by laarmada**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-15571

### Change description ###

upload docs and pay hearing fee should be removed when state is Decision Outcome or Final Orders issued

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by Gareth40343 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java [hunk 1]
```diff
@@ -56,4 +61,30 @@ private CallbackResponse changeState(CallbackParams callbackParams) {
     public List<CaseEvent> handledEvents() {
         return EVENTS;
     }
+
+    private void deleteNotificationAndInactiveTasks(CaseData caseData, String authToken) {
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java:65]: I see this kind of method being built a lot. Is there not a service / helper method we could put it in and call ?

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java [hunk 2]
```diff
@@ -39,9 +42,11 @@ protected Map<String, Callback> callbacks() {
 
     private CallbackResponse changeState(CallbackParams callbackParams) {
         if (featureToggleService.isCaseProgressionEnabled()) {
+            String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
             CaseData caseData = callbackParams.getCaseData().toBuilder()
                 .build();
             CaseData.CaseDataBuilder<?, ?> caseDataBuilder = caseData.toBuilder();
+            deleteNotificationAndInactiveTasks(caseData, authToken);
```
Comment (m-meulendijks-v1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java:49]: Should we not be doing this in the dashboard handlers:
uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/DefendantDecisionOutcomeDashboardHandler.java
uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/ClaimantDecisionOutcomeDashboardHandler.java

**Review by m-meulendijks-v1 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java (hunk 2 above)
Comment (laarmada) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java]: @m-meulendijks-v1 I have updated, please have a look. Thanks.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java (hunk 1 above)
Comment (laarmada) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DecisionOutcomeCallbackHandler.java]: @ss-evoco  I have change the way to do that, now I'm overriding a method, please have a look

**Review by ss-evoco — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Pull request merged**